### ISSUE TITLE
Add support for Basic Authentication in outbound HTTP requests

### ISSUE THREAD

**Issue by hodgestar** [praekelt/junebug#99]
Currently Junebug supports supplying URLs for sending messages, message status updates and channel status updates to. These URLs should allow a username and password to be specified within the URL and have those converted into Basic Authentication HTTP headers when HTTP requests are made to them.

E..g `https://user:pw@example.com/` should become a request to `https://example.com/` with `user` and `pw` supplied as the basic authentication username and password.

**Comment by imsickofmaps** [comment]
@rudigiesler can you point me at the code that needs fixing for this? Is it in Junebug (I couldn't see where mo_url is consumed in this repo) or deeper in Vumi?

When we try to use Basic Auth in mo_url we get stuff like the following in the logs:
 ```
File "/usr/local/lib/python2.7/site-packages/idna/core.py", line 253, in check_label
5/17/2017 3:59:57 PM    raise InvalidCodepoint('Codepoint {0} at position {1} of {2} not allowed'.format(_unot(cp_value), pos+1, repr(label)))
5/17/2017 3:59:57 PMidna.core.InvalidCodepoint: Codepoint U+003A at position 8 of u'junebug:password@api' not allowed
```

**Comment by rudigiesler** [comment]
@imsickofmaps The code that makes the http request is here: https://github.com/praekelt/junebug/blob/develop/junebug/workers.py#L282-L289 . We use `treq` for the request, and I'm guessing that treq doesn't support basic auth in the URL, or maybe we need to set a flag or something, we would investigate treq to see what can be done there.

If treq doesn't support it, then we need to extract the auth details from the URL, and pass the auth details and URL as separate fields to treq.

**Comment by imsickofmaps** [comment]
@rudigiesler properly supporting auth seems like a better solution for Junebug. Treq does support basic auth, as a param on the request: http://treq.readthedocs.io/en/latest/api.html#treq.request

**Comment by rudigiesler** [comment]
@imsickofmaps In that case, we'll need to extract the basic auth details from the URL, and pass the URL and auth details as separate parameters to treq.

**Comment by smn** [comment]
Implemented in #129
